tests/test_logging.py

```diff
@@ -26,25 +26,47 @@ def test_setOverrideLogLevel(self):
         self.assertEqual(notherlogger.getEffectiveLevel(), logging.ERROR)
         setOverrideLogLevel('INFO')
         somelogger = getLogger('foo.bar')
-        setOverrideLogLevel('INFO')
-
-
-class TestLogging2(TestCase):
-
-    def setUp(self):
-        initLogging()
 
     def test_getLevelName(self):
         self.assertEqual(getLevelName('ERROR'), logging.ERROR)
         self.assertEqual(getLevelName('FATAL'), logging.ERROR)
         self.assertEqual(getLevelName('OFF'), logging.CRITICAL)
+        setOverrideLogLevel('INFO')
```

Comment:
I was puzzled by this line before – now you moved it from one function to the other (because you moved that function from one class to the other).

@kba Please explain the reason for this in the first place!

Frankly, I don't know what unittest or pytest is supposed to do with our import-time state change. We have no explicit module or class or testcase setUp/tearDown (except for L34 before, which as I have argued should not have worked). But if pytest does in fact undo basicConfig between classes _somehow_, then at least this made _some_ sense.

Reply:
> @kba Please explain the reason for this in the first place!

I can't unfortunately. `git blame` didn't help. IIRC I used `tox` at the time to do cross-version testing and that behaved differently wrt to clearing module cache between tests or for python <= 2.7 or something like that.. The split into two test cases was also for that reason and is obsolete.